docs/compilers/Dynamic Analysis/MetadataFormat.md

```diff
@@ -0,0 +1,48 @@
+#### <a name="MethodSpansBlob"></a>Method Spans Blob
+Span is a quadruple of integers and a document reference:
+
+* Start Line
+* Start Column
+* End Line
+* End Column
+* Document
+
+The values of a span must satisfy the following constraints
+
+* Start Line is within range [0, 0x20000000).
+* End Line is within range [0, 0x20000000).
+* Start Column is within range [0, 0x10000)
+* End Column is within range [0, 0x10000)
+* End Line is greater or equal to Start Line.
+* If Start Line is equal to End Line then End Column is greater than Start Column.
+
+_Method Spans Blob_ has the following structure:
+
+    Blob ::= header span-record (span-record | document-record)*
+
+#####header
+| component        | value stored            | integer representation |
+|:-----------------|:------------------------|:-----------------------|
+| _InitialDocument_| Document row id         | unsigned compressed    |
+
+#####sequence-point-record
+| component      | value stored                                                 | integer representation                      |
+|:---------------|:-------------------------------------------------------------|:--------------------------------------------|
+| _ΔLines_       | _EndLine_ - _StartLine_                                      | unsigned compressed                         |
+| _ΔColumns_     | _EndColumn_ - _StartColumn_                                  | _ΔLines_ = 0: unsigned compressed, non-zero |
+|                |                                                              | _ΔLines_ > 0: signed compressed             |
+| _δStartLine_   | _StartLine_ if this is the first non-hidden sequence point   | unsigned compressed                         |
```

Comment:
Are we collecting coverage only for non hidden sequence points? If yes then it would mean we would not be able to show coverage for lines under #external or #line directives.
Additionally I think you do want to use the term sequence point here instead of span, right?